src/scales-factory.js

```diff
@@ -109,17 +109,52 @@ var scalesStrategies = {
         return func;
     },
 
-    ordinal: (varSet, props, interval) => {
+    ordinal: (varSet, props, interval, data) => {
```

Comment:
Maybe better use `ordinal: ({varSet, props, interval, data})`. This will make the scale interface more friendly.